examples/cpp/dds/AdvancedConfigurationExample/CMakeLists.txt

```diff
@@ -21,6 +21,10 @@ if(NOT fastcdr_FOUND)
     find_package(fastcdr REQUIRED)
```

Comment:
Regarding this specific commit I would like to have some more info. I understand that including the `EDP.h` header file requires to explicitly link against foonathan due to the private members defined in class `EDP`. As you have extended the mock classes in the test suite, this file has been added to the compilation of some specific tests which then have to link explicitly against foonathan. My question is if those other tests/examples that are not compiling this file require also to link against the dependency.

My concern is related with the examples. Fast DDS depends on foonathan to be built but that does not mean that the user application requires to link against foonathan (unless they are using some specific Fast DDS public API that is defined by foonathan). In fact, I suppose that Fast DDS should not expose foonathan API and that the foonathan API should be wrapped. This way, having a Fast DDS binary is enough and you would not require linking your application against a dependency that the user might not be interested in.

Reply:
I perfectly understand your point. In fact, it is a problem just on the windows platform (maybe `fastrtps` is not exporting something). In Linux/mac, neither tests nor examples were required to be linked against foonathan. 

What is more, the compilation error in windows was related with a missing foonathan's include (`#include <foonathan/memory/container.hpp>` due to the new `EDP.h`, like you said) and the solution afaik should not be directly linking to foonathan but extending the `CMake` with the right `include_directories(${foonathan_include_dirs})`.

For the moment, the only workaround that I found solved the issue was linking to foonathan in both: several examples and couple of test suites.

I am friendly pinging @MiguelCompany in case he could give us further insight or clue.